src/DataGeneration/bootstrap.jl

```diff
@@ -30,8 +30,8 @@ input2 = BootstrapInput{MovingBlock}(;kwargs...)
 """
 struct BootstrapInput{T<:TSBootMethod} <: DataGenInput
     input_data::Array{<:Real} # array of data to be resampled
-    n::Integer # desired size of resampled data
-    block_size::Float32 #desired average block size (will add more to this later)
+    n::Int64 # desired size of resampled data
+    block_size::Float64 #desired average block size (will add more to this later)
 
```

Comment:
I am sorry but this is not what you should be doing. My mistake if I was not clear enough. I agree with you that now, the structs fields are concretes, which is very good for performance. But you cannot use it with other Integers than Int64 now... which is not very good, what if you needed some BigInt someday ? 

Also, `input_data::Array{<:Real}` is not perfect since its type cannot be inffered from the struct. Try constructing your struct as follow: 

```julia
struct BootstrapInput{T<:TSBootMethod,TI,TF,TR} <: DataGenInput
    input_data::Array{TR} # array of data to be resampled
    n::TI # desired size of resampled data
    block_size::TF #
    function BootstrapInput{T,TI,TF,TR}(...)
        ....
        new(...)
    end
end
```
Then you may construct a generic constructor that infer the types from input data: 

```julia
function BootstrapInput(args)
    # compute T,TI,TF,TR from input arguments
    TF = typeof(block_size)
    TI = typeof(n)
    TR = eltype(input_data)
    # You could also enforce here using promotion rules that TF == TR and use one less type parameter. 
    # Then return the correctly typed object: 
    return BootstrapInput{T,TI,TF,TR}(...)
end
```

Then it would be possible to pass through your code with weird types that you did not expect before -- e.g. automatic differentiation by ForwardDiff, or BigFloats for precision, or GPU CUDA floats, without modifying your code.

Reply:
Oh, this is wayyy better than what I was doing! I'll do this to the rest of the structs tonight.

Reply:
More generally, you may read https://docs.julialang.org/en/v1/manual/performance-tips/

In particular, sections "Avoid containers with abstract type parameters" is a more precise and well-explained version of what I just said.